File: tests/analyzer/sourcecode/benign/threat-runtime-obfuscation-steganography.py
# Benign regression test for threat-runtime-obfuscation-steganography.
# A stego library reading a PNG, but the only call named exec is a method call
# (node.exec), preceded by a dot. The bare builtin is what matters, so this
# must not trigger.
from stegano import lsb

img = lsb.reveal("cover.png")
node.exec(payload)

File: tests/analyzer/sourcecode/benign/threat-runtime-obfuscation-steganography.js
// Benign regression test for the image-obfuscation runtime detection rule.
// Buffer.concat() and a PNG asset path sit near Playwright's page.$eval
// selector method, which is not the bare execution builtin, so this must
// not trigger.
const icon = './assets/logo.png';
const combined = Buffer.concat([a, b]);

async function readCount(page) {
  return page.$eval('#count', (el) => el.textContent);
}
